feat: v7.23.0

sethkfman opened the pull request:
## **Description**

Release 7.23.0 PR

[Release Tracker Sheet](https://docs.google.com/spreadsheets/d/1tsoodlAlyvEUpkkcNcbZ4PM9HuC9cEM80RZeoVv5OCQ/edit#gid=1301559947&range=A3)

sethkfman committed:
bump version 7.23.0 - 1327

sethkfman committed:
updated changelog

github-actions[bot] committed:
chore: cherry-pick #9614 (#9615)

This PR cherry-picks #9614

Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>

sethkfman committed:
version bump 1328 RC - 2

Cal-L committed:
fix: Fix/rn svg pod files (#9630)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR updates the pod files related to RN SVG update PR -
https://github.com/MetaMask/metamask-mobile/pull/9595
Since the issue was fixed in the previous PR, it's most likely because
it was patched on the JS side.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/9594

## **Manual testing steps**

Refer to original PR

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

github-actions[bot] committed:
chore: cherry-pick #9204 (#9636)

This PR cherry-picks #9204

Co-authored-by: Jony Bursztyn <jonathan@bursztyn.io>
Co-authored-by: EtherWizard33 <eric.lamontagne@consensys.net>

katkopel commented:
**Testing iOS v.7.23.0 (1330)**


**Onboarding: custom ETH mainnet**
✅ Custom mainnet Scenario 1 (Verifying Custom Mainnet Screen Appears):
✅ Custom mainnet Scenario 2 (Entering an Invalid Mainnet RPC URL):
✅ Custom mainnet Scenario 3 (Entering a valid mainnet RPC URL):
✅ Custom mainnet Scenario 4 (Editing a custom mainnet URL on the settings and privacy view):
✅ Custom mainnet Scenario 5 (Deleting a custom mainnet URL on the settings and privacy view):
https://drive.google.com/file/d/1oowqJddZYzcXASpAgdXB6yDumX0baKe6/view 


**QR Scanner Regression**
✅ From the in-app camera I should be able to scan a valid QR code containing an address:
✅ I should be able to use the QR scanner on the send flow to scan an address:
https://drive.google.com/file/d/1SAZWppsaHUsJhzoY68SbZCCQfifwFR0z/view 

✅ When I scan a dapp QR code using the in-app camera I should see the appropriate warning message:
✅ When I scan a dapp QR code using the QR scanner on the send flow I should see the appropriate error message:
https://drive.google.com/file/d/1TMw-fSjHbfyy6pKHyHd8OeTpUqh0Zepf/view 

✅ When I scan an invalid address QR code using the in app camera I should see the appropriate warning message:
❌ When I scan an invalid address QR code using the QR scanner on the send flow I should see the appropriate warning message: Wrong error message appears
https://drive.google.com/file/d/1GaZPEBKgoV1MevkBepDbQ_-BVfFMvUXb/view 

✅ A user can scan a private key QR code using the in-app camera to import an account:
https://drive.google.com/file/d/1EvJD9SUT7xJrchvTiHs0aa3O6XMmRXiW/view 

✅ User scans MMM Receive QR code on different network:
❗Chain ID 1 not found error when scanning Eth receive QR from another network
https://drive.google.com/file/d/16riGqB1QWlGvgk2sk_6RwnLvSoy6FFvV/view 


**Setting: General view**
✅ Scenario: Changing the device language from English to Russian:
❗  “Your wallet is ready” welcome screen is in English
https://drive.google.com/file/d/1DNGFwYK0NR7N4t4G_I4FE7b0K_Ha6jSf/view 

✅ Scenario: Changing the app language from English to Filipino:
https://drive.google.com/file/d/1c8u08rqCzuNILTvzPu6bSAXQ569s4tmj/view 


**Browser Regression**
✅ Scenario: Searching an invalid url and prompts the user with an error message:
✅ Scenario: Navigation with browser controls:
✅ Scenario: ENS website loads correctly:
✅ Scenario: From the browser I can successfully upload a picture:
✅ Scenario: Address bar:
✅ Scenario: Adding browser Favorites:
✅ Scenario: Browser Options Menu:
Browser back button - 
Share, Network change in wallet, & New Tab - 
✅ Scenario: Visiting a malicious website prompts the user with the phishing detection warning:
❌ Scenario: Navigate to opensea.io, rarible.com, pancakeswap.finance, app.uniswap.org, and connect wallet account
	❗ Opensea.io connect redirects to app store to download MetaMask
https://drive.google.com/file/d/1ERfHpJix2VqVODykyDTpk-tcz4PpkOoD/view
OpenSea issue - https://drive.google.com/file/d/10kS5-PhYKudGcvs0UW7UK1_9an1DDq3g/view  


**Network regression**
✅ Scenario: I can remove a custom network (Palm) that was added via the popular network flow:
✅ Scenario: I can remove a custom network that was added via the custom network flow.
✅ Scenario Outline: Adding a network via a Dapp:
✅ Scenario Outline: Adding a network via the new popular network flow:
✅ Scenario: Adding a network via the custom network flow:
✅ Scenario: Adding a custom network and editing the network:
✅ Scenario: Editing a popular network:
https://drive.google.com/file/d/1iUN5PEbCR9kONyOfQJZNXRqafu951QG6/view

github-actions[bot] committed:
chore: cherry-pick #9428 (#9641)

This PR cherry-picks #9428

Co-authored-by: MetaMask Bot <37885440+metamaskbot@users.noreply.github.com>
Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: sethkfman <seth.kaufman@consensys.net>

Cal-L committed:
chore: cherry-pick-Chore/1723 fix broken sign (#9660)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution? -->

The PR that introduced permission controller middleware
https://github.com/MetaMask/metamask-mobile/pull/9521, broke verifying
personal sign on the test dapp. This PR resolves a few areas in the app
that needed to use lowercased account addresses. Another note: Since
permission middleware now handles `eth_accounts`, we migrated a
condition to use the correct origin based on SDK, WC, or dapp in the
`BackgroundBridge`. This code is a temporary fix and should be
refactored in the future. To keep this PR smaller, we can follow up with
removing `eth_accounts` from RPCMethodMiddleware in another PR.

## **Related issues**

Fixes: [#1723](https://github.com/MetaMask/mobile-planning/issues/1723),
https://app.zenhub.com/workspaces/mobile-release-testing-6249e5242464b70013315a98/issues/gh/metamask/metamask-mobile/9628

## **Manual testing steps**

Before fix:
1. Connect to the MM test dapp
2. Trigger personal sign
3. Trigger verify
4. Results will not appear

After fix:
1. Connect to the MM test dapp
2. Trigger personal sign
3. Trigger verify
4. Results should show address used to sign
5. Switch/connect another account
6. Trigger personal sign
4. Results should show address used to sign

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->



https://github.com/MetaMask/metamask-mobile/assets/10508597/e38a71b5-6ac0-40ea-b3de-36b38830e7c9


### **After**

<!-- [screenshots/recordings] -->

Signing verification works even after switching accounts


https://github.com/MetaMask/metamask-mobile/assets/10508597/5c43af3b-7c9f-456b-a870-a5345027642b

WC connects and is able to trigger a transaction



https://github.com/MetaMask/metamask-mobile/assets/10508597/7009ac14-fa87-4ebd-88e0-022f2f30264e




## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

github-actions[bot] committed:
chore: cherry-pick #9612 (#9689)

This PR cherry-picks #9612

Co-authored-by: CW <chris.wilcox@consensys.net>

sethkfman committed:
chore: version bump 1331 (#9691)

## **Description**

This PR bumps the 7.23.0 version to 1331 build.

## **Related issues**

Fixes: NA

## **Manual testing steps**

1. NA
3.

## **Screenshots/Recordings**

NA

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

Jonathansoufer committed:
chore: fix cherry conflict while merging JSON fix. (#9716)

## **Description**

This is a cherry picking of commit
6c5f55c3407b26f42d270c26cff21883ee0a2e31 from PR #9568

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/9559

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

Cal-L committed:
fix: cherry-pick - Move permission middleware to be later than rpc method middleware (#9699) (#9717)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR cherry picks PR #9699 

## **Related issues**

fixes #9443 
fixes MetaMask/mobile-planning#1796

## **Manual testing steps**

Refer to original PR

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [X] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [X] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [X] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

github-actions[bot] committed:
chore: cherry-pick #9720 (#9722)

This PR cherry-picks #9720

Co-authored-by: EtherWizard33 <165834542+EtherWizard33@users.noreply.github.com>

github-actions[bot] committed:
chore: cherry-pick #9733 (#9737)

This PR cherry-picks #9733

Co-authored-by: Cal Leung <cleun007@gmail.com>

sethkfman committed:
chore: bump RC 4 1332 (#9738)

## **Description**

Chore bump release version

## **Related issues**

Fixes: NA

## **Manual testing steps**

1. NA
2.
3.

## **Screenshots/Recordings**

NA

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

github-actions[bot] committed:
chore: cherry-pick #9764 (#9765)

This PR cherry-picks #9764

Co-authored-by: Owen Craston <owen.craston@consensys.net>

NicolasMassart committed:
bump build number to 1333 (#9769)

- Bump build number to 1333
- Updated the Flask number to the same value as discussed with
@sethkfman

sethkfman committed:
fix: e2e builds (#9773)

## **Description**

This PR upgrades the Ruby version to be compatible with E2E test
machines.

## **Related issues**

Fixes: NA

## **Manual testing steps**

1. NA

## **Screenshots/Recordings**

NA

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

github-actions[bot] committed:
chore: cherry-pick #9767 (#9776)

This PR cherry-picks #9767

Co-authored-by: Jony Bursztyn <jonathan@bursztyn.io>
Co-authored-by: David Walsh <davidwalsh83@gmail>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>

SamuelSalas commented:
QA Sign-off for v7.23.0 build 1333

sethkfman committed:
merge main

sethkfman committed:
revert .yarnrc commit

sethkfman committed:
remove duplicate code

NicolasMassart reviewed (approved):
Looks good to me.
